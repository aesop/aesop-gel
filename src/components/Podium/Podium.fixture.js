export default {
  className: undefined,
  id: 'id',
  isActive: true,
  paddingTop: 'none',
  paddingBottom: 'none',
  transition: 'fade',
};
