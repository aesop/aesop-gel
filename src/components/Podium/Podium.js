import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { ThemeContextProvider } from '~/contexts';
import Transition from '~/components/Transition';
import styles from './Podium.module.css';

const Podium = forwardRef(function PodiumRef(
  {
    backgroundColor,
    children,
    className,
    'data-test-ref': dataTestRef,
    horizontalPadding,
    isActive,
    isActiveOnMount,
    isHorizontalFlushOnLarge,
    isHorizontalFlushOnMedium,
    isHorizontalFlushOnSmall,
    paddingBottom,
    paddingLeft,
    paddingRight,
    paddingTop,
    theme,
    transition,
    verticalPadding,
  },
  ref,
) {
  const classSet = cx(
    styles.base,
    styles[`${paddingBottom ? paddingBottom : verticalPadding}PaddingBottom`],
    styles[`${paddingLeft ? paddingLeft : horizontalPadding}PaddingLeft`],
    styles[`${paddingRight ? paddingRight : horizontalPadding}PaddingRight`],
    styles[`${paddingTop ? paddingTop : verticalPadding}PaddingTop`],
    { [styles.horizontalFlushOnSmall]: isHorizontalFlushOnSmall },
    { [styles.horizontalFlushOnMedium]: isHorizontalFlushOnMedium },
    { [styles.horizontalFlushOnLarge]: isHorizontalFlushOnLarge },
    className,
  );

  const style = backgroundColor ? { backgroundColor } : undefined;

  return (
    <ThemeContextProvider theme={theme}>
      <Transition isActiveOnMount={true} type={transition}>
        <section className={classSet} data-test-ref={dataTestRef} style={style}>
          <Transition
            isActive={isActive}
            isActiveOnMount={isActiveOnMount}
            type={transition}
          >
            <div ref={ref}>{children}</div>
          </Transition>
        </section>
      </Transition>
    </ThemeContextProvider>
  );
});

Podium.propTypes = {
  backgroundColor: PropTypes.string,
  children: PropTypes.node,
  className: PropTypes.string,
  'data-test-ref': PropTypes.string,
  horizontalPadding: PropTypes.oneOf(['none', 'small']),
  isActive: PropTypes.bool,
  isActiveOnMount: PropTypes.bool,
  isHorizontalFlushOnLarge: PropTypes.bool,
  isHorizontalFlushOnMedium: PropTypes.bool,
  isHorizontalFlushOnSmall: PropTypes.bool,
  paddingBottom: PropTypes.oneOf(['none', 'small', 'medium', 'large']),
  paddingLeft: PropTypes.oneOf(['none', 'small']),
  paddingRight: PropTypes.oneOf(['none', 'small']),
  paddingTop: PropTypes.oneOf(['none', 'small', 'medium', 'large']),
  theme: PropTypes.oneOf(['dark', 'light']),
  transition: PropTypes.oneOf([
    'fade',
    'shiftInDown',
    'shiftInLeft',
    'slideDown',
    'slideRight',
    'slowFade',
    'zoom',
  ]),
  verticalPadding: PropTypes.oneOf(['none', 'small', 'medium', 'large']),
};

Podium.defaultProps = {
  backgroundColor: undefined,
  children: undefined,
  className: undefined,
  'data-test-ref': undefined,
  horizontalPadding: 'none',
  isActive: true,
  isActiveOnMount: undefined,
  isHorizontalFlushOnLarge: undefined,
  isHorizontalFlushOnMedium: undefined,
  isHorizontalFlushOnSmall: undefined,
  paddingBottom: undefined,
  paddingLeft: undefined,
  paddingRight: undefined,
  paddingTop: undefined,
  theme: undefined,
  transition: undefined,
  verticalPadding: 'none',
};

export default Podium;
