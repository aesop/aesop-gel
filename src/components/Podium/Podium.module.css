@import 'styles/index';

.base {
  display: block;
  padding: 0;
}

.smallPaddingTop {
  padding-top: 50px;
}

.mediumPaddingTop {
  padding-top: 100px;

  @media (--viewport-lg) {
    padding-top: 150px;
  }
}

.largePaddingTop {
  padding-top: 150px;

  @media (--viewport-lg) {
    padding-top: 250px;
  }
}

.smallPaddingBottom {
  padding-bottom: 50px;
}

.mediumPaddingBottom {
  padding-bottom: 100px;

  @media (--viewport-lg) {
    padding-bottom: 150px;
  }
}

.largePaddingBottom {
  padding-bottom: 150px;

  @media (--viewport-lg) {
    padding-bottom: 250px;
  }
}
