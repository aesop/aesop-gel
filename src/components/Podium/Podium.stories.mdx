import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import { withKnobs } from '@storybook/addon-knobs';
import { select, text } from '@storybook/addon-knobs';
import CarouselFixture from '~/components/Carousel/Carousel.fixture';
import Carousel from '~/components/Carousel';
import Podium from './Podium';
import PodiumFixture from './Podium.fixture';

<Meta title="Components/Podium" component={Podium} decorators={[withKnobs]} />

# Podium

Presented with both `paddingTop` and `paddingBottom` set to `small`.

<Preview>
  <Story name="Base component">
    <div style={{ border: '1px solid #333', margin: '20px'}}>
      <Podium
        paddingTop={select(
          'paddingTop',
          ['none', 'small', 'medium', 'large'],
          'small',
        )}
        paddingBottom={select(
          'paddingBottom',
          ['none', 'small', 'medium', 'large'],
          'small',
        )}
      >
        <Carousel
          introduction={CarouselFixture.introduction}
          slides={CarouselFixture.slides}
        />
      </Podium>
    </div>
  </Story>
</Preview>

## API

<Props of={Podium} />
