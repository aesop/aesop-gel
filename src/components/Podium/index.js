/* istanbul ignore file */
import { Podium } from './Podium';

export { Podium };
