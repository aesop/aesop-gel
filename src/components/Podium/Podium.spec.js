import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Podium from './Podium';
import PodiumFixture from './Podium.fixture';

configure({ adapter: new Adapter() });

describe('<Podium />', () => {
  it('should be defined', () => {
    expect(Podium).toBeDefined();
  });

  it('renders base component correctly', () => {
    const tree = renderer.create(<Podium />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
