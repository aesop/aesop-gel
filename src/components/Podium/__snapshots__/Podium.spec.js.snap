// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Podium /> renders base component correctly 1`] = `
<section
  className="base nonePaddingTop nonePaddingBottom"
/>
`;
